### ISSUE TITLE
Why does box_w only divide by strides [-1] 

### ISSUE THREAD

**Issue by fengxiaoming520** [yjh0410/pytorch-yolo-v2-v3#25]
Hello, I would like to ask you a question:
In the multi_gt_creator function of the tools.py file, why does box_w only divide by strides [-1] before calculating iou, does this not always cause iou to be calculated on the minimum or maximum downsampling multiple? Instead of calculating at all downsampling multiples.

`
box_ws = box_w / strides[-1]  
`
`
box_hs = box_h / strides[-1]  
`

**Comment by yjh0410** [comment]
It is because that my all anchor boxes have been divided by 32(the largest stride). Therefore, before computing IoU, I have to divide the groundtruth box by 32. Even though some anchor boxes correspond to stride =8 or 16, see the formulations:
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;tw&nbsp;=&nbsp;np.log(box_ws&nbsp;/&nbsp;p_w)
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;th&nbsp;=&nbsp;np.log(box_hs&nbsp;/&nbsp;p_h)



It eliminates the effects of factor stride meaning that we actually needn't to divide the anchor boxes and groundtruth bxoes by stride and just omit strides. If I didn't divide my all anchor boxes by 32, I didn't need to divide the box_w and box_h by 32, either.


------------------&nbsp;原始邮件&nbsp;------------------
发件人:&nbsp;"fengxiaoming520"<notifications@github.com&gt;;
发送时间:&nbsp;2020年3月21日(星期六) 晚上10:45
收件人:&nbsp;"yjh0410/pytorch-yolo-v2-v3"<pytorch-yolo-v2-v3@noreply.github.com&gt;;
抄送:&nbsp;"Subscribed"<subscribed@noreply.github.com&gt;;
主题:&nbsp;[yjh0410/pytorch-yolo-v2-v3] Why does box_w only divide by strides [-1]  (#25)





 
Hello, I would like to ask you a question:
 In the multi_gt_creator function of the tools.py file, why does box_w only divide by strides [-1] before calculating iou, does this not always cause iou to be calculated on the minimum or maximum downsampling multiple? Instead of calculating at all downsampling multiples.
 
box_ws = box_w / strides[-1]  
 box_hs = box_h / strides[-1]

**Comment by fengxiaoming520** [comment]
Let ground truth find the anchor box that suits it, and then use the anchor box to decide which downsampling multiple it belongs to. Do you mean that?

**Comment by yjh0410** [comment]
Yes. In my configuration, there are 9 anchor boxes where the first three anchor boxes correspond to stride=8, the middle three ones correspond to stride=16, and the last three ones correspond to stride=32. If the one of first three anchor boxes is chosen，then the groundtruth is allocated to stride=8 feature map.




------------------&nbsp;原始邮件&nbsp;------------------
发件人:&nbsp;"fengxiaoming520"<notifications@github.com&gt;;
发送时间:&nbsp;2020年3月21日(星期六) 晚上11:11
收件人:&nbsp;"yjh0410/pytorch-yolo-v2-v3"<pytorch-yolo-v2-v3@noreply.github.com&gt;;
抄送:&nbsp;"ら ．Secret"<1394571815@qq.com&gt;;"Comment"<comment@noreply.github.com&gt;;
主题:&nbsp;Re: [yjh0410/pytorch-yolo-v2-v3] Why does box_w only divide by strides [-1]  (#25)





 
Let ground truth find the anchor box that suits it, and then use the anchor box to decide which downsampling multiple it belongs to. Do you mean that?

**Comment by fengxiaoming520** [comment]
I see, thank you for the explanation.^v^

**Comment by yjh0410** [comment]
You are welcome ！！




------------------&nbsp;原始邮件&nbsp;------------------
发件人:&nbsp;"fengxiaoming520"<notifications@github.com&gt;;
发送时间:&nbsp;2020年3月21日(星期六) 晚上11:23
收件人:&nbsp;"yjh0410/pytorch-yolo-v2-v3"<pytorch-yolo-v2-v3@noreply.github.com&gt;;
抄送:&nbsp;"ら ．Secret"<1394571815@qq.com&gt;;"Comment"<comment@noreply.github.com&gt;;
主题:&nbsp;Re: [yjh0410/pytorch-yolo-v2-v3] Why does box_w only divide by strides [-1]  (#25)





 
I see, thank you for the explanation.^v^

**Comment by Haoyue-code** [comment]
![image](https://github.com/yjh0410/yolov2-yolov3_PyTorch/assets/76245900/82b65185-536f-44ec-a52c-f89b232b127f)
您好，请问我在运行demo的时候出现这个问题怎么解决呢
